GVFS/GVFS.Service/GvfsService.cs

```diff
@@ -335,5 +347,20 @@ private void LogExceptionAndExit(Exception e, string method)
             this.tracer.RelatedError(metadata, "Unhandled exception in " + method);
             Environment.Exit((int)ReturnCode.GenericError);
         }
+
+        private void EnableAccessToAuthenticatedUsers(string rootDirectory)
+        {
+            // GVFS Config is written to a temporary file and then renamed to its final destination.
+            // For this rename operation to succeed, user needs to have delete permission on the 
+            // destination file, in case it is pre-existing. If the pre-existing file was created 
+            // by a different user, then the delete will fail.
+            // Reference: https://stackoverflow.com/questions/22107812/privileges-owner-issue-when-writing-in-c-programdata
+            // This work around allows safe write to succeed in C:\ProgramData directory.
+
+            DirectorySecurity security = Directory.GetAccessControl(Path.GetDirectoryName(this.serviceDataLocation));

```

Comment:
@jrbriggs are we sure this is all correct behavior here?

Reply:
No specific issues, just general uneasiness about changing the permissions on a directory. We've done it before and then regretted it, though I agree that this seems unavoidable if we want to store the config file in this location. Have we done enough to rule out other locations?

Reply:
The only folder I can think of that might not need special permissions grants and would be a singleton per machine is %PUBLIC%. However, I'm not sure I feel much better about putting VFS for Git config in that folder than I do granting these permissions to %PROGRAMDATA%/GVFS/GVFS.Service. Thoughts?